docs/rest-api-v1-v2-transition-guide.md

```diff
@@ -0,0 +1,120 @@
+# REST-API V1 V2 - Transition Guide
+
+## What Changed
+Assisted service is now using a V2 API, which introduces a new resource type called `infra_env`.
+`infra_env` represents the configuration needed to create the discovery ISO.
+Hosts will now register to `infra_env`, unlike in V1 API, where hosts registered directly to a `cluster`.
+
+This change comes to support more advanced use-cases, such as late-binding, in which:
```

Comment:
@avishayt @romfreiman should we expose it in this doc? we didn't plan to support late binding in the cloud, the api enable it but i'm not sure we should explain the details here.

Reply:
Why not mention it?  The SaaS UI doesn't expose it, but the REST API can be used directly either in the SaaS or podman.  If we don't want to allow it in the REST API then we should block it.